src/vue/components/panel-compact.vue

```diff
@@ -110,78 +122,137 @@
-                 v-if="settings.contentOptions.items"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.contentOptions.items.verboseMode"
-                 @change="contentOptionChanged"
-                 v-if="settings.contentOptions.items"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.uiOptions.items.skin"
-                 @change="uiOptionChanged"
-                 v-if="settings.uiOptions && settings.uiOptions.items"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.uiOptions.items.panel"
-                 @change="uiOptionChanged"
-                 v-if="settings.uiOptions && settings.uiOptions.items && app.isDevMode()"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.uiOptions.items.popup"
-                 @change="uiOptionChanged"
-                 v-if="settings.uiOptions && settings.uiOptions.items"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.uiOptions.items.panelOnActivate"
-                 @change="uiOptionChanged"
-                 v-if="settings.uiOptions && settings.uiOptions.items"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="languageSetting"
-                 :key="languageSetting.name"
-                 @change="resourceSettingChanged"
-                 v-for="languageSetting in resourceSettingsLexicons"></setting>
-        <setting :classes="['alpheios-panel__options-item']" :data="languageSetting"
-                 :key="languageSetting.name"
-                 @change="resourceSettingChanged"
-                 v-for="languageSetting in resourceSettingsLexiconsShort"></setting>
-
-        <setting :classes="['alpheios-panel__options-item']"
-                 :data="settings.contentOptions.items.enableWordUsageExamples" @change="contentOptionChanged"
-                 v-if="settings.contentOptions.items"></setting>
-
-        <setting :classes="['alpheios-panel__options-item']"
-                 :data="settings.contentOptions.items.wordUsageExamplesAuthMax"
-                 @change="contentOptionChanged"
-                 v-if="settings.contentOptions.items"></setting>
-
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.contentOptions.items.wordUsageExamplesMax"
-                 @change="contentOptionChanged"
-                 v-if="settings.contentOptions.items"></setting>
-
-        <setting :classes="['alpheios-panel__options-item']"
-                 :data="settings.contentOptions.items.enableLemmaTranslations" @change="contentOptionChanged"
-                 v-if="settings.contentOptions.items"></setting>
-
-        <setting :classes="['alpheios-panel__options-item']" :data="settings.contentOptions.items.locale"
-                 @change="contentOptionChanged"
-                 v-if="settings.contentOptions.items"></setting>
+        <setting
```

Comment:
You have removed all checks for settings. Don't you think that it could break all the code if some parts of settings won't  be loaded?

Reply:
I think that since we already have the check for data validity in `setting.vue` (https://github.com/alpheios-project/components/blob/comp-i365/src/vue/components/setting.vue#L2) there is no reason to duplicate it here. It's probably better to group all data check inside a component, if possible, because this way we can have it in one place instead of 30+ places if they be put where the component instances are referred. What do you think?

Reply:
I am not sure, but I think that before rendering the component the v-if check will be done.
And if some property in the chain is undefined, it won't render and check validity inside settings.vue.
So I фь тще ыгку that those checks could be interchangeable.

Reply:
If data passed to `settings.vue` UI component is incorrect, the component will not render, and an app will handle that smoothly, without throwing any errors. Did some tests to confirm that. I think this is what we want from it.

If some property along the chain does not exist, this will throw an error, and that's bad. But I think there is no reason to add an execution-time check for such condition: settings data is set statically, so if there are any errors here (i.e. some prop in settings does not exist), we should catch them during testing. Then it will never happen during app's execution. It's not some user-entered or other third-party data that requires dynamic checking. With static data there is no reason to increase code complexity and waste CPU cycles on dynamic checks, on my opinion. @balmas, what do you think?

Reply:
I think I agree with @kirlat here but also am not sure what scenario "If some property along the chain does not exist" represents. Can you give an example?

Reply:
>"If some property along the chain does not exist"
I see here several variants:
   settings property is rather complex and includes different sub-settings loaded from different sources - from json files, from localStorage or sessionStorage
   so we could change the code later and won't catch that there are some problems with settings
   the extension or embed-lib could be used with some browser, that won't have some parts of it
   
    If to be honest, I am not sure that we should to be so afraid of v-if here, v-if has its own advantages and setting is really a div+input. 
    And of course you both could have other opinions and be right, but I believe that as javascript is not a strict language we shouldn't avoid to check variables types and existance. And our goal is not skip the errors but be able to see them.

    But of course it is only my opinion :)

Reply:
Also there is a description when we should v-if and v-show
https://vuejs.org/v2/guide/conditional.html#v-if-vs-v-show

And from this information - settings are good for v-if and really won't be rendered ever (if we have problems with some settings), will decrease initial load.
So I don't agree that using v-if here could lead to some CPU load or other difficulties.

Reply:
It's hard for me to estimate how possible are the scenarios described. I'm not against `v-if`s, if used properly. What I don't like about the version we had before was the repetition of a prop name:
```
<setting
            :classes="['alpheios-panel__options-item']"
            :data="settings.uiOptions.items.panelOnActivate"
            v-if="settings.uiOptions.items.panelOnActivate"
            @change="uiOptionChanged"
        >
```
we had to type the property name `settings.uiOptions.items.panelOnActivate` twice for each settings component.

I agree that JS is a dynamic language and certain run time checks are necessary, but I think it would be better if we put them either inside the `settings` component or inside a settings object.
We probably need to update settings object itself anyway to integrate it with Vuex that we've started to use. That will make working with settings simpler. But I'm not sure it is a high priority. @balmas, what do you think?

Reply:
we have much work to do with settings. And we need at some point to enable server-side storage of user preferences, which I think requires some refactoring first.  I don't think that will make the initial release of the MVP but am not sure yet.